builder/vmware/common/driver_workstation9.go

```diff
@@ -215,3 +231,112 @@ func (d *Workstation9Driver) ToolsInstall() error {
 func (d *Workstation9Driver) GetVmwareDriver() VmwareDriver {
 	return d.VmwareDriver
 }
+
+// checkNetmapConfExists checks if the network mapper configuration file exists.
+// If not, if attempts to generate the file using generateNetmapConfig.
+func checkNetmapConfExists() (NetworkNameMapper, error) {
+	pathNetmap := workstationNetmapConfPath()
+
+	// Check if the default network mapper configuration file exists.
+	if _, err := os.Stat(pathNetmap); os.IsNotExist(err) {
+		log.Printf("A network mapper configuration file does not exist in the default path: %s", pathNetmap)
+
+		// The file does not exist, check the alternate configuration path.
+		libpath, _ := workstationVMwareRoot()
+		pathNetworking := filepath.Join(libpath, "networking")
+		log.Printf("Checking alternate path for network mapper configuration file: %s", pathNetworking)
+		if _, err := os.Stat(pathNetworking); err == nil {
+			// Alternate networking configuration path exists. Using this path.
+			log.Printf("Located the network mapper configuration file: %s", pathNetworking)
+			fd, err := os.Open(pathNetworking)
+			if err != nil {
+				return nil, err
+			}
+			defer fd.Close()
+
+			// Then we pass the handle to the networking configuration parser.
+			return ReadNetworkingConfig(fd)
+		} else if runtime.GOOS == "windows" {
+			// Neither the default nor alternate exists.
+			// Generate the network mapper configuration file in the default path.
+			log.Printf("Generating a network mapper configuration file in the default path: %s", pathNetmap)
+			pathNetmap, err := generateNetmapConfig()
+			if err != nil {
+				log.Fatalf("error generating the absent network configuration file: %v", err)
+				return nil, err
+			}
+			// If the file exists, attempt to read the newly generated
+			// network mapper configuration file.
+			return ReadNetmapConfig(pathNetmap)
+		} else {
+			// For non-Windows, if neither the default nor alternate
+			// configuration paths exist, return an error.
+			return nil, fmt.Errorf("error determining network mappings from files in path: %s", libpath)
+		}
+	} else if err != nil {
+		// There was an error other than the file not existing.
+		return nil, err
+	}
+
+	// If the default network mapper configuration file exists, read the configuration.
+	log.Printf("Located the network mapper configuration file: %s", pathNetmap)
+	return ReadNetmapConfig(pathNetmap)
```

Comment:
Instead of the nested if statements let's see if we can write this to return early when there is no more work to be done. We can also wrap the underlying error so that we can check above if it is an os.ErrNotExist error as a toggle for calling the generate function on Windows. 
For example 


```go
        // Check if the default network mapper configuration file exists.
	_, err := os.Stat(pathNetmap)
	if err == nil {
		// If the default network mapper configuration file exists, read the configuration.
		log.Printf("Located the network mapper configuration file: %s", pathNetmap)
		return ReadNetmapConfig(pathNetmap)
	}
	if !os.IsNotExist(err) {
		return nil, fmt.Errorf("error determining network mappings from files %w", err)
	}

	log.Printf("A network mapper configuration file does not exist in the default path: %s", pathNetmap)
	// The file does not exist, check the alternate configuration path.
	libpath, _ := workstationVMwareRoot()
	pathNetworking := filepath.Join(libpath, "networking")
	log.Printf("Checking alternate path for network mapper configuration file: %s", pathNetworking)
	_, err = os.Stat(pathNetworking)
	if err != nil { // If we get an error we wrap the error. The caller can check if it is a ErrNotExist and act accordingly
		return nil, fmt.Errorf("error determining network mappings from files %w", err)
	}

	// Alternate networking configuration path exists. Using this path.
	log.Printf("Located the network mapper configuration file: %s", pathNetworking)
	fd, err := os.Open(pathNetworking)
	if err != nil {
		return nil, err
	}
	defer fd.Close()

	// Then we pass the handle to the networking configuration parser.
	return ReadNetworkingConfig(fd)
```

Reply:
On it - will update and send back soon.

Reply:
Resolved in https://github.com/hashicorp/packer-plugin-vmware/commit/2ebbe20b26e25aa4be8dd12291b20fc408d291ff and tested!